### PR TITLE
request to merge assignment 3 into develop branch

### PR DESCRIPTION
**Pull request by munendra-successive**
1.create two components: a parent component that acts as a provider and a child component that consumes the context.
Create a context to manage a user's authentication status (logged in or out).
Implement a login button in the child component that, when clicked, updates the authentication status in the context to "logged in."
Display a message in the child component based on the user's authentication status. If the user is logged in, show "Welcome, [username]!" Otherwise, display "Please log in."

2.Extend the previous application to demonstrate the use of nested contexts.
Create a new context to manage user preferences (e.g., theme preference).
Modify the parent component to provide both the authentication and preferences contexts.
Create a child component that consumes both contexts.
Allow the user to change their theme preference (light/dark) using a button in the child component.
Display the theme preference in the UI and adjust the component's styling accordingly.

3.creat pages: Home and About.
Implement routing using React Router DOM.
Create navigation links to switch between the Home and About pages.
Display appropriate content on each page.
Add a "404 Not Found" page for any invalid routes.

4.Enhance the previous application by adding protected routes that require authentication.
Create a login page with a form that accepts a username and password.
Implement authentication logic (e.g., hardcode a username and password for now).
Secure the About page so that it can only be accessed by authenticated users.
Redirect unauthenticated users to the login page.
Display a message on the Home page welcoming the authenticated user.

5.Extend the application to include nested routes for a more complex user interface.
Create a new section of your application (e.g., a Dashboard) with multiple sub-pages (e.g., Dashboard, Profile, Settings).
Implement nested routing within this section using nested routes.
Add navigation links to switch between the sub-pages within the Dashboard section.
Customize the content displayed on each sub-page.

6.Create a list of items (e.g., products, articles) in your application.
Implement dynamic routing to display details for each item when clicked.
Create a route parameter that represents the item's ID in the URL (e.g., "/products/:id").
Fetch item details based on the route parameter and display them on the detail page.
Add a "Go Back" button on the detail page to return to the list.

### CODE REVIEW

File: src/assignment-3/Question-1/Parent.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+1.create two components: a parent component that acts as a provider and a child component that consumes the context.
+Create a context to manage a user's authentication status (logged in or out).
+Implement a login button in the child component that, when clicked, updates the authentication status in the context to "logged in."
+Display a message in the child component based on the user's authentication status. If the user is logged in, s
+how "Welcome, [username]!" Otherwise, display "Please log in."
+
+*/
+
+export const UserAuthenticate = createContext();
```
Comment (UrvashiChaudhary) [inline src/assignment-3/Question-1/Parent.jsx:13]: please provide context a meaningful name

File: src/assignment-3/Question-2/Parent.jsx
Comment (UrvashiChaudhary) [inline src/assignment-3/Question-2/Parent.jsx:1]: provide meaningful name to the context

**Review by UrvashiChaudhary — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/user-event": "13.5.0",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-router-dom": "^6.18.0",
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply package.json:11]: Removed

File: src/assignment-3/Question-1/Child.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { useContext, useState } from "react";
+import { UserAuthenticate } from "./Parent";
+
+const Child = () => {
+  const { setLogin, islogin, userDetails } = useContext(UserAuthenticate);
+  const [username, setUsername] = useState();
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-3/Question-1/Child.jsx:6]: Define initial value as empty string

File: src/assignment-3/Question-1/Parent.jsx (hunk 1 above)
Comment (munendra-successive) [reply src/assignment-3/Question-1/Parent.jsx:13]: Define initial value as empty string

File: src/assignment-3/Question-2/Parent.jsx
Comment (munendra-successive) [reply src/assignment-3/Question-2/Parent.jsx:1]: Renamed UserAuth as UserAuthContext
Renamed UserPref as UserPrefContext

File: src/App.js [hunk 4]
```diff
@@ -1,97 +1,158 @@
...
+  HomeFive,
+  NavbarFive,
+  ProfileFive,
+  ServiceFive,
+  SettingFive,
+  HomeSix,
+  ProductDetail,
+} from "./assignment-3";
+
+export const UserAuth = createContext();
```
Comment (UrvashiChaudhary) [inline src/App.js:31]: provide appropriate name to the context

**Review by UrvashiChaudhary — CHANGES_REQUESTED** [review]

File: src/assignment-3/Question-6/ProductDetail.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+const ProductDetail = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  return (
+    <div id="product">
+      <h4>Product Details Page</h4>
+
+      {ProdDesc.map((item) => (
+        <div>
+          {item.id == id ? (
```
[the remark this answers is not in the document]
Comment (munendra-successive) [reply src/assignment-3/Question-6/ProductDetail.jsx]: Done

File: src/App.js (hunk 4 above)
Comment (munendra-successive) [reply src/App.js:31]: Renamed UserAuth to UserAuthContext

**Pull request merged**